pkg/webhooks/machineset_webhook_test.go

```diff
@@ -216,22 +225,107 @@ func TestMachineSetCreation(t *testing.T) {
 			expectedError: "[providerSpec.template: Required value: template must be provided, providerSpec.workspace: Required value: workspace must be provided, providerSpec.network.devices: Required value: at least 1 network device must be provided]",
 		},
 		{
-			name:         "with vSphere and the template, workspace and network devices set",
-			platformType: osconfigv1.VSpherePlatformType,
-			clusterID:    "vsphere-cluster",
+			name:         "with PowerVS and with correct data",
+			platformType: osconfigv1.PowerVSPlatformType,
+			clusterID:    "powervs-cluster",
 			providerSpecValue: &runtime.RawExtension{
-				Object: &machinev1.VSphereMachineProviderSpec{
-					Template: "template",
-					Workspace: &machinev1.Workspace{
-						Datacenter: "datacenter",
-						Server:     "server",
+				Object: &powervsmachine.PowerVSMachineProviderConfig{
+					ServiceInstanceID: "TestServiceInstanceID",
+					KeyPairName:       "TestKeyPair",
+					Image: powervsmachine.PowerVSResourceReference{
+						Name: pointer.StringPtr("TestNetworkName"),
 					},
-					Network: machinev1.NetworkSpec{
-						Devices: []machinev1.NetworkDeviceSpec{
-							{
-								NetworkName: "networkName",
-							},
-						},
```

Comment:
why are we modifying the existing testcase?

Reply:
Fixed it in the latest commit